src/osd/osd_types.cc

```diff
@@ -3606,15 +3606,6 @@ void object_copy_data_t::decode(bufferlist::iterator& bl)
     ::decode(data, bl);
     ::decode(omap_data, bl);
     ::decode(cursor, bl);
-    if (struct_v >= 2)
-      ::decode(omap_header, bl);
-    if (struct_v >= 3) {
-      ::decode(snaps, bl);
-      ::decode(snap_seq, bl);
-    } else {
```

Comment:
dachary, thank you for reviewing. 

in next patch i will resolve your comments.